{
  "uid" : "4aa62630150afe09",
  "name" : "Attempt to retrieve a non-existent identity -- @1.1 ",
  "fullName" : "REST API interaction with TrustPoint: Attempt to retrieve a non-existent identity",
  "historyId" : "a3e8cd088ca177d7853a3edafd2eb932",
  "time" : {
    "start" : 1746705354253,
    "stop" : 1746705354283,
    "duration" : 30
  },
  "status" : "failed",
  "statusMessage" : "AssertionError: Step not implemented: API client authentication.\n",
  "statusTrace" : "  File \"/home/runner/work/trustpoint/trustpoint/.venv/lib/python3.12/site-packages/behave/model.py\", line 1909, in run\n    match.run(runner.context)\n  File \"/home/runner/work/trustpoint/trustpoint/.venv/lib/python3.12/site-packages/behave/matchers.py\", line 104, in run\n    self.func(context, *args, **kwargs)\n  File \"trustpoint/features/steps/common_steps.py\", line 101, in step_api_client_authenticated\n    raise AssertionError(msg)\n",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ ],
  "testStage" : {
    "status" : "failed",
    "statusMessage" : "AssertionError: Step not implemented: API client authentication.\n",
    "statusTrace" : "  File \"/home/runner/work/trustpoint/trustpoint/.venv/lib/python3.12/site-packages/behave/model.py\", line 1909, in run\n    match.run(runner.context)\n  File \"/home/runner/work/trustpoint/trustpoint/.venv/lib/python3.12/site-packages/behave/matchers.py\", line 104, in run\n    self.func(context, *args, **kwargs)\n  File \"trustpoint/features/steps/common_steps.py\", line 101, in step_api_client_authenticated\n    raise AssertionError(msg)\n",
    "steps" : [ {
      "name" : "Given an API client is authenticated",
      "time" : {
        "start" : 1746705354255,
        "stop" : 1746705354255,
        "duration" : 0
      },
      "status" : "failed",
      "statusMessage" : "AssertionError: Step not implemented: API client authentication.\n",
      "statusTrace" : "  File \"/home/runner/work/trustpoint/trustpoint/.venv/lib/python3.12/site-packages/behave/model.py\", line 1909, in run\n    match.run(runner.context)\n  File \"/home/runner/work/trustpoint/trustpoint/.venv/lib/python3.12/site-packages/behave/matchers.py\", line 104, in run\n    self.func(context, *args, **kwargs)\n  File \"trustpoint/features/steps/common_steps.py\", line 101, in step_api_client_authenticated\n    raise AssertionError(msg)\n",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : true,
      "hasContent" : true
    }, {
      "name" : "When the API client sends a GET request to \"/api/identities/unknown123\"",
      "time" : {
        "start" : 1746705354283,
        "stop" : 1746705354283,
        "duration" : 0
      },
      "status" : "skipped",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : false
    }, {
      "name" : "Then the API response should have a status code of 404",
      "time" : {
        "start" : 1746705354283,
        "stop" : 1746705354283,
        "duration" : 0
      },
      "status" : "skipped",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : false
    }, {
      "name" : "And the response payload should include an error message stating \"Identity not found\"",
      "time" : {
        "start" : 1746705354283,
        "stop" : 1746705354283,
        "duration" : 0
      },
      "status" : "skipped",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : false
    } ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 4,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : true
  },
  "afterStages" : [ ],
  "labels" : [ {
    "name" : "severity",
    "value" : "normal"
  }, {
    "name" : "feature",
    "value" : "REST API interaction with TrustPoint"
  }, {
    "name" : "framework",
    "value" : "behave"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ {
    "name" : "identifier",
    "value" : "unknown123"
  } ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Product defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "history" : {
      "statistic" : {
        "failed" : 2,
        "broken" : 0,
        "skipped" : 0,
        "passed" : 0,
        "unknown" : 0,
        "total" : 2
      },
      "items" : [ {
        "uid" : "27fd5d13ee38cebf",
        "reportUrl" : "https://Trustpoint-Project.github.io/trustpoint/16/#testresult/27fd5d13ee38cebf",
        "status" : "failed",
        "statusDetails" : "AssertionError: Step not implemented: API client authentication.\n",
        "time" : {
          "start" : 1746616905432,
          "stop" : 1746616905462,
          "duration" : 30
        }
      } ]
    },
    "tags" : [ ]
  },
  "source" : "4aa62630150afe09.json",
  "parameterValues" : [ "unknown123" ]
}